### PR TITLE
Allowing scrub configs begin_day/end_day to include 7 and begin_hour/end_hour to include 24 is confusing

### PR DESCRIPTION
**Pull request by dzafman**
Initial version ready for review

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by dzafman** [comment]
Testing of this pull request with #37908 passed:

https://pulpito.ceph.com/dzafman-2020-11-05_19:46:57-rados-wip-zafman-testing-48077-47930-distro-basic-smithi/

**Comment by dzafman** [comment]
@neha-ojha 

In the doc they refer to "osd scrub begin week day" and "osd scrub end week day" with spaces instead of underscores.  My change that you requested I have the underscores.  Should I change to spaces?

`osd_scrub_begin_week_day=0 osd_scrub_end_week_day=0 
`

**Comment by neha-ojha** [comment]
> @neha-ojha
> 
> In the doc they refer to "osd scrub begin week day" and "osd scrub end week day" with spaces instead of underscores. My change that you requested I have the underscores. Should I change to spaces?
> 
> `osd_scrub_begin_week_day=0 osd_scrub_end_week_day=0 `

yeah, it'd be nice to keep it consistent

File: doc/rados/configuration/osd-config-ref.rst [hunk 1]
```diff
@@ -218,37 +218,54 @@ scrubbing operations.
...
 
 :Description: This restricts scrubbing to days of the week earlier than this.
-              0 or 7 = Sunday, 1 = Monday, etc.
-:Type: Integer in the range of 0 to 7
-:Default: ``7``
+              0 = Sunday, 1 = Monday, etc.  Use ``osd scrub begin week day = 0``
+              and ``osd scrub end week day = 0`` for the entire week.
+              Along with ``osd scrub begin week day``, they define a time
+              window, in which the scrubs can happen. But a scrub will be performed
+              But a scrub will be performed
```
Comment (neha-ojha) [inline doc/rados/configuration/osd-config-ref.rst:264]:
```suggestion
              window, in which the scrubs can happen. But a scrub will be performed
              
```

File: qa/standalone/scrub/osd-scrub-test.sh [hunk 2]
```diff
@@ -344,6 +342,49 @@ function TEST_deep_scrub_abort() {
...
+    create_pool $poolname 1 1
+    wait_for_clean || return 1
+
+    # Trigger a scrub on a PG
+    local pgid=$(get_pg $poolname SOMETHING)
+    local primary=$(get_primary $poolname SOMETHING)
+    local last_scrub=$(get_last_scrub_stamp $pgid)
+    ceph tell $pgid scrub $(( 24 * 60 * 60 - 10)) || return 1
+
+    # Scrub should not run
```
Comment (neha-ojha) [inline qa/standalone/scrub/osd-scrub-test.sh:377]: can you please add an explanation for this and why scrub should not run for future readers?

File: doc/rados/configuration/osd-config-ref.rst [hunk 3]
```diff
@@ -218,37 +218,54 @@ scrubbing operations.
 
 ``osd scrub begin hour``
 
-:Description: The time of day for the lower bound when a scheduled scrub can be
-              performed.
-:Type: Integer in the range of 0 to 24
+:Description: This restricts scrubbing to this hour of the day or later.
+              Use ``osd scrub begin hour = 0`` and ``osd scrub end hour = 0``
+              for the entire day.  Along with ``osd scrub end hour``, they define a time
```
Comment (neha-ojha) [inline doc/rados/configuration/osd-config-ref.rst:223]: nit: here and everywhere
```suggestion
              to allow scrubbing the entire day.  Along with ``osd scrub end hour``, they define a time
```

**Review by neha-ojha — APPROVED** [review]
looks good overall, left a few nits

**Comment by neha-ojha** [comment]
@dzafman Seeing TEST_scrub_extented_sleep failures most likely due to this PR https://pulpito.ceph.com/teuthology-2020-11-12_07:01:02-rados-master-distro-basic-smithi/5615390/

**Comment by dzafman** [comment]
> @dzafman Seeing TEST_scrub_extented_sleep failures most likely due to this PR https://pulpito.ceph.com/teuthology-2020-11-12_07:01:02-rados-master-distro-basic-smithi/5615390/

The TEST_scrub_extented_sleep test was broken from the original implementation.  See my fixes for that and the ver-health.sh: ceph/ceph#38054

**Pull request merged**